File: modul_6/PRAK604-2510817220008-NamiraHumaira.py
jumlah_pagar = 0
jumlah_bintang = 0
kode_shikamaru = input()
kode_musuh = input()
if len(kode_shikamaru) != len(kode_musuh):
    print("Panjang kalimat berbeda, pesan palsu")
    exit()
for i in range(len(kode_shikamaru)):
    if kode_shikamaru[i] == ' ' and kode_musuh[i] == ' ':
        continue
    if kode_musuh[i] == kode_shikamaru[i]:
        print("*", end="")
        jumlah_bintang += 1
    else:
        print("#", end="")
        jumlah_pagar += 1
print("\n* = ", jumlah_bintang)
print("# = ",  jumlah_pagar)
if (jumlah_bintang >= jumlah_pagar): print("Pesan Asli")
else: print("Pesan Palsu")

File: modul_6/PRAK604-2510817220008-NamiraHumaira.c
#include <stdio.h>
#include <string.h>
int main()
{
    char pesan_sikamaru[100],pesan_rahasia[100];
    int bintang=0,pagar=0;
    fgets(pesan_sikamaru, sizeof(pesan_sikamaru), stdin);
    fgets(pesan_rahasia, sizeof(pesan_rahasia), stdin);
    pesan_sikamaru[strcspn(pesan_sikamaru, "\n")]= '\0';
    pesan_rahasia[strcspn(pesan_rahasia, "\n")]= '\0';
    if(strlen(pesan_sikamaru) == strlen(pesan_rahasia))
    {
        int panjang_pesan = strlen(pesan_sikamaru);
        for(int i=0; i<panjang_pesan; i++)
        {
            if(pesan_sikamaru[i]==' ' && pesan_rahasia[i]==' ')
            {
                continue;
            }
            if(pesan_sikamaru[i]==pesan_rahasia[i])
            {
                printf("*");
                bintang++;
            }  
            else
            {
                printf("#");
                pagar++;
            }
        }
        printf("\n* = %d",bintang);
        printf("\n# = %d",pagar);
        if(bintang>=pagar)
        {
            printf("\nPesan Asli");
        }
        else
        {
            printf("\nPesan Palsu");
        }
    }
    else
    {
        printf("Panjang Kalimat berbeda, pesan palsu");
    }
    return 0;
}